dxr/plugins/clang/dxr-index.cpp

```diff
@@ -1087,7 +1095,7 @@ class IndexConsumer : public ASTConsumer,
             text[i] != '\t' && text[i] != '\n')
           text[i] = '?';
       }
-      recordValue("text", text, true);
+      macromap.insert(make_pair(nameStart, text));
```

Comment:
If it is indeed a problem, one alternative would be to build the map in the post-processing global phase, when memory is more plentiful.